text/0000-macros-1.1.md

```diff
@@ -0,0 +1,473 @@
+- Feature Name: `rustc_macros`
+- Start Date: 2016-07-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Stabilize a very small sliver of today's procedural macro system in the
+compiler, just enough to get basic features like custom derive working. Ensure
+that the features stabilized here will not pose a maintenance burden on the
+compiler but also don't try to stabilize enough features for the "perfect macro
+system" at the same time. Overall, this should be considered an incremental
+step towards an official "macros 2.0".
+
+# Motivation
+[motivation]: #motivation
+
+Some large projects in the ecosystem today, such as [serde] and [diesel],
+effectively require the nightly channel of the Rust compiler. Although most
+projects have an alternative to work on stable Rust, this tends to be far less
+ergonomic and comes with its own set of downsides, and empirically it has not
+been enough to push the nightly users to stable as well.
+
+[serde]: https://github.com/serde-rs/serde
+[diesel]: http://diesel.rs/
+
+These large projects, however, are often the face of Rust to external users.
+Common knowledge is that fast serialization is done using serde, but to others
+this just sounds likes "fast Rust needs nightly". Over time this persistent
+thought process creates a culture of "well to be serious you require nightly"
+and a general feeling that Rust is not "production ready".
+
+The good news, however, is that this class of projects which require nightly
+Rust almost all require nightly for the reason of procedural macros. Even
+better, the full functionality of procedural macros is rarely needed, only
+custom derive! The purpose of this RFC, as a result, is to provide these crates
```

Comment:
Might be nice to note why derive is a nice sub-case of macros - hygiene is not really an issue, modularisation is much easier.